src/chocolatey/infrastructure.app/registration/ContainerBinding.cs

```diff
@@ -126,6 +129,19 @@ public void RegisterComponents(Container container)
                   return list.AsReadOnly();
               },
               Lifestyle.Singleton);
+
+            container.Register<IEnumerable<IValidation>>(
+                () =>
+                {
+                    var list = new List<IValidation>
+                    {
+                        new GlobalConfigurationValidation(),
+                        new SystemStateValidation(container.GetInstance<IPendingRebootService>())
+                    };
+
+                    return list.AsReadOnly();
+                },
+                Lifestyle.Singleton);

```

Comment:
Validations Engine is an entirely different commit. I would separate the validation engine from the reboot detection.

I would also separate the commits for Pending Reboot detection versus Package Exit Code indicating Reboot.

Reply:
For global validations, let's even create an issue to tie those commits to. It's an interesting addition and we want to be able to tell folks when we first implemented global validations. Much harder to do if there is not an issue covering it.